### DISCUSSION TITLE
Using navigation plugin with dynamically created pages

### DISCUSSION THREAD

**Discussion by jamesgrubb** [11ty/eleventy#1477] [Q&A]
Hello, I'm using the navigation plugin. As the plugin relies on data in hardcoded (I believe) into the frontmatter how can I use the plugin with pages that are created through data i.e dynamically where I have no control offer the front matter apart from setting the tags? Hope that makes sense?

**Comment by Ryuno-Ki** [comment]
I'd suggest to open this on the [eleventy-navigation](https://github.com/11ty/eleventy-navigation) repo as issue (feature request?).

**Comment by renestalder** [comment]
I have the same case, but as I have full control over the data in my data file where I actually make my API requests, I found there is no need for the navigation plugin and that I can build a navigation tree by myself. And if your API is a CMS that is flexible enough with custom models, you can even integrate the navigation building into the CMS and retrieve the navigation data from the API.
